packages/live-preview-sdk/src/inspectorMode/utils.ts

```diff
@@ -127,115 +89,131 @@ function isSameElement(a: AutoTaggedElement, b: AutoTaggedElement): boolean {
-  // The fastest way to look up & iterate over DOM. Ref:
-  // https://stackoverflow.com/a/2579869
-  // Initialize a TreeWalker to traverse all nodes, using a filter function to determine which nodes to consider.
-  // Terminology:
-  // FILTER_SKIP: Skip the current node
-  // FILTER_REJECT: Skip the current node and all its children
-  const walker = document.createTreeWalker(root, NodeFilter.SHOW_ALL, (node) => {
-    // Ignore tags that are not visible on the page
-    if (
-      node.nodeType === Node.ELEMENT_NODE &&
-      IGNORE_TAGS.includes((node as HTMLElement).tagName)
-    ) {
-      return NodeFilter.FILTER_REJECT;
+  const element = child.parentElement;
+  if (element) {
+    const counters = {
+      sameInformation: 0,
+      tagged: 0,
+      all: 0,
+    };
+
+    let sibling = child.nextElementSibling;
+    while (sibling) {
+      const siblingSourceMap = decode(sibling.textContent || '');
+      if (siblingSourceMap) {
+        counters.tagged += 1;
+        if (siblingSourceMap.href === sourceMap.href) {
+          counters.sameInformation = +1;
+        }
+      }
+      counters.all += 1;
+      sibling = sibling.nextElementSibling;
     }
 
-    // Detect auto-tagged nodes
-    const sourceMap = getSourceMap(node);
-    if (sourceMap?.origin === 'contentful.com') {
-      // Ignore if the parent is already tagged with the same information
-      if (
-        (isTaggedElement(node.parentElement) || hasTaggedParent(node.parentElement, sourceMap)) &&
-        !ignoreManual
-      ) {
-        return NodeFilter.FILTER_SKIP;
-      }
-      return NodeFilter.FILTER_ACCEPT;
+    const parentInformation = getParent(element, sourceMap, limit - 1);
+    if (parentInformation && parentInformation.counters.sameInformation >= 1) {
+      return parentInformation;
     }
 
-    // For non-text nodes, if ignoreManual is true, skip manually tagged elements.
-    if (ignoreManual) {
-      return NodeFilter.FILTER_SKIP;
+    if (counters.sameInformation >= 1) {
+      // at least one more siblings has the same information
+      return { element, counters };
     }
+  }
 
-    // Accept the node if it is tagged according to the isTaggedElement function, else skip.
-    return isTaggedElement(node) ? NodeFilter.FILTER_ACCEPT : NodeFilter.FILTER_SKIP;
-  });
+  return null;
+}
 
-  const taggedElements: Element[] = [];
-  const elementsForTagging: AutoTaggedElement<Element>[] = [];
+function findStegaNodes(container: HTMLElement) {
+  let baseArray: HTMLElement[] = [];
+  if (typeof container.matches === 'function' && container.matches('*')) {
+    baseArray = [container];
+  }
 
-  // Iterate over the nodes accepted by the TreeWalker.
-  while (walker.nextNode()) {
-    const node = walker.currentNode;
+  return [...baseArray, ...Array.from(container.querySelectorAll<HTMLElement>('*'))]
+    .filter((node) => !node.matches('script, style, meta, title'))
```

Comment:
You can use `container.querySelectorAll(":not(script,style,meta,title)")` instead to avoid the `.filter` call

Reply:
You're right, saves some additional ms

<img width="1095" alt="Screenshot 2024-02-26 at 14 08 03" src="https://github.com/contentful/live-preview/assets/3918488/2d4f4c66-8181-4bcf-aaae-2e3305b98007">